python/src/aiconfig/default_parsers/openai.py

```diff
@@ -74,40 +49,66 @@ def serialize(
         Returns:
             str: Serialized representation of the prompt and inference settings.
         """
-        # Serialize Prompt input
-        prompt_input = (
-            data["prompt"] if isinstance(data["prompt"], str) else {"data": data["prompt"]}
-        )
+        serialized_prompts = []
 
-        # Cosntruct model settings, does not include prompt
-        prompt_model_metadata = {key: value for key, value in data.items() if key != "prompt"}
+        # Combine conversation data with any extra keyword args
+        conversation_data = {**data} | kwargs
 
-        # check if config already has model settings in global metadata
-        model_name = prompt_model_metadata.get("model", self.id())
-        global_model_metadata = ai_config.metadata.models.get(model_name, {})
+        if not "messages" in conversation_data:
+            raise ValueError("Data must have `messages` array to match openai api spec")
 
-        if global_model_metadata:
-            # Check if the model settings from the input data are the same as the global model settings
-            # Compute the difference between the global model settings and the model settings from the input data
-            # If there is a difference, then we need to add the different model settings as overrides on the prompt's metadata
-            override_keys = set(prompt_model_metadata.keys()) - set(global_model_metadata.keys())
+        # Extract the global settings for the model
+        global_model_settings = ai_config.get_global_settings(self.id())
 
-            override_settings = {key: prompt_model_metadata[key] for key in override_keys}
+        if global_model_settings:
+            # Identify the settings that differ from global settings
+            override_settings = {
+                key: copy.deepcopy(conversation_data[key])
+                for key in conversation_data
+                if key not in global_model_settings
+            }
 
-            if override_settings:
-                model_metadata = {"name": model_name, "settings": override_settings}
-            else:
-                model_metadata = model_name
+            model_metadata = ModelMetadata(**{"name": self.id(), "settings": override_settings})
         else:
-            model_metadata = {"name": model_name, "settings": prompt_model_metadata}
-
-        return Prompt(
-            name=prompt_name,
-            input=prompt_input,
-            metadata=PromptMetadata(model=PromptMetadata(**model_metadata)),
-            parameters=parameters,
-            **kwargs
-        )
+            model_metadata = ModelMetadata(
+                **{"name": self.id(), "settings": copy.deepcopy(conversation_data)}
+            )
+        # Remove messages array from model metadata. Handled separately
+        model_metadata.settings.pop("messages", None)
+
+        system_prompt = ""
+        user_prompt = ""
```

Comment:
This is set to a string but then set to a Prompt object below, which is confusing. Maybe just set to None?